File: DSA-Problems/0026-remove-duplicates-from-sorted-array/solution.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var removeDuplicates = function(nums) {
    const newArr = new Set(nums)

    nums.length = 0;
    newArr.forEach((value)=> {
        nums.push(value)
    })

   return nums.length
};

File: DSA-Problems/0026-remove-duplicates-from-sorted-array/solution.cpp
class Solution {
public:
    int removeDuplicates(vector<int>& nums) {

        //optimal solution: Two pointer approach

        int n = nums.size();
        if(n==0){
            return 0;
        }

        int index = 1; // position for next unique elements
        for(int i =1;i<n; i++){
            if(nums[i] != nums[i-1]){
                nums[index] = nums[i];
                index++;
            }
        }
        return index;
        
    }
};
